components/supabase/supabase.app.mjs

```diff
@@ -24,18 +25,143 @@ export default {
       label: "Row Data",
       description: "Enter the column names and values as key/value pairs",
     },
+    filter: {
+      type: "integer",
+      label: "Filter",
+      description: "Filter type for the query",
+      options: constants.FILTER_OPTIONS,
+    },
   },
   methods: {
     async _client() {
       return createClient(`https://${this.$auth.subdomain}.supabase.co`, this.$auth.service_key);
     },
-    async selectRow(table, column, value) {
+    async selectRow(opts) {
       const client = await this._client();
-      const { data } = await client
+      const args = {
+        table: opts.table,
+        column: opts.column,
+        value: opts.value,
+        orderBy: opts.orderBy,
+        ascending: opts.sortOrder === "ascending"
+          ? true
+          : false,
+        max: opts.max,
+      };
+      const filterMethod = opts.filter
+        ? this[constants.FILTER_METHODS[opts.filter]]
+        : null;
+      const response = opts.filter
+        ? await filterMethod(client, args)
+        : await this.getMaxRows(client, args);
+      return response.data;
+    },
+    async getMaxRows(client, {
+      table, orderBy, ascending, max,
+    }) {
+      return client
         .from(table)
         .select()
-        .eq(column, value);
-      return data;
+        .order(orderBy, {
+          ascending,
+        })
+        .limit(max);
+    },
+    async equalTo(client, {
+      table, column, value, orderBy, ascending, max,
+    }) {
+      return client
+        .from(table)
+        .select()
+        .eq(column, value)
+        .order(orderBy, {
+          ascending,
+        })
+        .limit(max);
```

Comment:
I think we could centralize the common parts here, since the only thing that changes between each of these is the filtering method. How about something like:

```
async baseFilterStart(client, { table }) {
  return client
    .from(table)
    .select();
},
async baseFilterEnd(obj, {
  ascending, max, orderBy
}) {
  return obj
    .order(orderBy, {
      ascending,
    })
    .limit(max);
},
```

So the equalTo method, for example, would be changed to only the unique filtering method:
```suggestion
    async equalTo(obj, {
      column, value,
    }) {
      return obj
        .eq(column, value);
```

So the response would be obtained by calling baseFilterStart, equalTo and baseFilterEnd, each with the result of the previous method. If all of these have to be promises (and not just the initial query), this succession of three methods would require chaining `.then` or `await`ing each of them.

But I think it's best to keep the shared logic in shared methods - what do you think? @michelle0927